### ISSUE TITLE
Enumerable of Enum doesn't work if followed by Values

### ISSUE THREAD

**Issue by usr-sse2** [commandlineparser/commandline#617]
I have such options:
```csharp
[Option("fm",  Separator=',', Default = new[] { FMode.S })]  
public IEnumerable<FMode> Mode { get; set; }
	 
[Option('q')]
public bool q { get;set; }
	 
[Value(0)]
public IList<string> Files { get; set; }
```

If I run my application with `--fm D,C a.txt` arguments, it says that option 'fm' is defined with a bad format. If I put any option between 'fm' and the value list, for example, `--fm D,C -q a.txt`, it is parsed successfully. You can check it in a fiddle: https://dotnetfiddle.net/yt5L9K

**Comment by rmunn** [comment]
This is caused by the fact that IEnumerable options try to grab every subsequent non-option argument that follows them, regardless of whether it makes sense, which is why it's grabbing the `a.txt` and trying to parse that as an FMode enum. I believe that this is non-intuitive behavior and that the way IEnumerable options should work should be changed as follows:

* If there is no separator given, grab all available non-option args
* If a separator is given, but the next non-option arg doesn't contain the separator, continue to grab all available non-option args
* If a separator is given, and the next non-option arg contains the separator, STOP and don't grab any further values (**this would be new behavior**)

Currently the code follows the first two of those rules, but not the third.

**Comment by rmunn** [comment]
This will be fixed by #684 once it's merged in.
